examples/controls_example/public/react_controls/timeslider_control/get_timeslider_control_factory.tsx

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { BehaviorSubject, debounceTime, first, map } from 'rxjs';
+import { EuiInputPopover } from '@elastic/eui';
+import {
+  apiHasParentApi,
+  apiPublishesDataLoading,
+  getViewModeSubject,
+  useBatchedPublishingSubjects,
+  ViewMode,
+} from '@kbn/presentation-publishing';
+import { ControlFactory } from '../types';
+import {
+  TimesliderControlState,
+  TimesliderControlApi,
+  TIMESLIDER_CONTROL_TYPE,
+  Services,
+  Timeslice,
+} from './types';
+import { initializeDefaultControlApi } from '../initialize_default_control_api';
+import { TimeSliderPopoverButton } from './components/time_slider_popover_button';
+import { TimeSliderPopoverContent } from './components/time_slider_popover_content';
+import { initTimeRangeSubscription } from './init_time_range_subscription';
+import {
+  FROM_INDEX,
+  roundDownToNextStepSizeFactor,
+  roundUpToNextStepSizeFactor,
+  TO_INDEX,
+} from './time_utils';
+import { initTimeRangePercentage } from './init_time_range_percentage';
+import './components/index.scss';
+import { TimeSliderPrepend } from './components/time_slider_prepend';
+
+export const getTimesliderControlFactory = (
+  services: Services
+): ControlFactory<TimesliderControlState, TimesliderControlApi> => {
+  return {
+    type: TIMESLIDER_CONTROL_TYPE,
+    getIconType: () => 'search',
+    getDisplayName: () =>
+      i18n.translate('controlsExamples.timesliderControl.displayName', {
+        defaultMessage: 'Timeslider',
+      }),
```

Comment:
I know this isn't used anywhere yet, but for anything in the UI, this should have a space:

```suggestion
      i18n.translate('controlsExamples.timesliderControl.displayName', {
        defaultMessage: 'Time slider',
      }),
```